sklearn/utils/validation.py

```diff
@@ -1091,17 +1099,28 @@ def is_sparse(dtype):
                 % (n_features, array.shape, ensure_min_features, context)
             )
 
-    # With an input pandas dataframe or series, we know we can always make the
-    # resulting array writeable:
-    # - if copy=True, we have already made a copy so it is fine to make the
-    #   array writeable
-    # - if copy=False, the caller is telling us explicitly that we can do
-    #   in-place modifications
-    # See https://pandas.pydata.org/docs/dev/user_guide/copy_on_write.html#read-only-numpy-arrays
-    # for more details about pandas copy-on-write mechanism, that is enabled by
-    # default in pandas 3.0.0.dev.
-    if _is_pandas_df_or_series(array_orig) and hasattr(array, "flags"):
-        array.flags.writeable = True
+    if force_writeable:
+        array_data = array.data if sp.issparse(array) else array
+        copy_params = {"order": "K"} if not sp.issparse(array) else {}
```

Comment:
default if "C" for the `copy` method of ndarray while it's "K" for `np.copy`. But I use the method because `np.copy` doesn't give the expected result on sparse arrays.